scripts/weatherfx.js

```diff
@@ -29,16 +30,13 @@ Hooks.once('ready', async function () {
     if (fvttVersion < 10) {
         particleWeather = 'fxmaster.updateWeather'
     }
-    let playlist = game.playlists?.contents.find((p) => p.name === playlistName);
-    let playlistExists = playlist ? true : false;
-    if (!playlistExists) await weatherfxPlaylist(playlistName);
+    await weatherfxPlaylistExists();
 });
 
 Hooks.on('canvasReady', async function () {
-    if (canvas.scene.getFlag("weatherfx", "active"))
-        if (enableSound)
-            if (canvas.scene.getFlag("weatherfx", "audio") != "")
-                AudioHelper.play({ src: canvas.scene.getFlag("weatherfx", "audio"), volume: 0.8, loop: true }, true);
+    if (await canvas.scene.getFlag("weatherfx", "active") !== undefined || await canvas.scene.getFlag("weatherfx", "audio") !== undefined || await canvas.scene.getFlag("weatherfx", "currentWeather") !== undefined)
```

Comment:
This is because the flag will exist only in scenes that used the older version of the module. so we get ride of those.

Reply:
regarding the undefined condition did you tried to test it differently? when I did in other case it didn't go well, maybe it is foundry or maybe its me recalling wrong.

Reply:
If get Flag returns a value it will have a value if not it will return undefined (assuming the get Flag function actually returns undefined)

Reply:
OK only for this one though:
```js
canvas.scene.getFlag("weatherfx", "active")
```
I need to check if this is `!== undefined` because the value inside might be boolean `false`.

Reply:
Boolean false is the same as the value returned from the function equals undefined